### PR TITLE
[AIRFLOW-3282] Azure Kubernetes Service Operator

### PR DESCRIPTION
**Pull request by apraovjr**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3282
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
       An Azure Kubernetes Service Operator which allows you to run AKS

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/aks_operator.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        Default to Linux.
+    :type os_type: str or OSType Enum
+    :param tags: Resource tags.
+    :type tags: dict[str, str]
+    :param location: Required resource location
+    :type location: str
+
+    :Example:
+
+    >>> a = AzureContainerInstancesOperator(task_id="task",ci_conn_id='azure_kubernetes_default',
```
Comment (elizabethhalper) [inline airflow/contrib/operators/aks_operator.py]: Can you change this to AzureKubernetesKubernetesOperator?

Comment (omusavi) [reply airflow/contrib/operators/aks_operator.py]: Also why is it AzureKubernetes**Kubernetes**Operator. Shouldnt it just be AzureKubernetesOperator?

**Review by elizabethhalper — APPROVED** [review]
This looks good! Just a few things to clean up but otherwise great.

**Review by omusavi — APPROVED** [review]
Looks good besides the name of the operator being weird (see other comment)

File: airflow/contrib/operators/aks_operator.py (hunk 1 above)
Comment (apraovjr) [reply airflow/contrib/operators/aks_operator.py]: @omusavi made change

**Review by pjlittle — APPROVED** [review]

**Pull request merged**